### PR TITLE
small changes for named tuples 

### PR DESCRIPTION
**Pull request by palashahuja**

### CODE REVIEW

File: pgmpy/factors/Factor.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-        marginalize_index = np.array(np.where(np.in1d(factor.scope(), variables)))
+        marginalize_index = np.array(np.where(np.array([i in variables for i in factor.scope()])))
```
Comment (jaidevd) [inline pgmpy/factors/Factor.py]: `marginalize_index` is only being used for indexing, right? So in the interest of better performance, simply doing

``` python
marginalize_index = np.array([i in variables for i in factor.scope()])
```

will create a boolean array which can itself be used for indexing. No need for the extra `np.where` and `np.array` calls.

File: pgmpy/factors/Factor.py [hunk 2]
```diff
@@ -194,7 +194,7 @@ def marginalize(self, variables, inplace=True):
...
         assign = np.array(factor.cardinality)
```
Comment (jaidevd) [inline pgmpy/factors/Factor.py]: Ignore the comment above if 
```
assign.shape[0] != marginalize_index.shape[0]
```

File: pgmpy/factors/Factor.py [hunk 3]
```diff
@@ -194,7 +194,7 @@ def marginalize(self, variables, inplace=True):
...
         assign[marginalize_index] = -1
```
Comment (jaidevd) [inline pgmpy/factors/Factor.py:199]: Ignore the comment above if 

``` python
assign.shape[0] != marginalize_index.shape[0]
```

and feel free to delete both comments.

**Comment by palashahuja** [comment]
@jaidevd The shapes of `assign` and `marginalize_index` must be same.

**Comment by jaidevd** [comment]
Let's see if something breaks.

**Comment by jaidevd** [comment]
LGTM.

**Comment by ankurankan** [comment]
@palashahuja Are there more changes to the state thing (that we had discussed) or this is it?

**Comment by palashahuja** [comment]
@ankurankan There are some more changes. Please don't merge this branch right now. I will keep updating the pull request.

**Comment by ankurankan** [comment]
@palashahuja Then please don't work on the state representation thing because I have already started fixing it.

**Comment by palashahuja** [comment]
@ankurankan It's almost done.... I have to still look for the errors in bayesian model(file)...

**Comment by abinashpanda** [comment]
@palashahuja Could you please have a look at the test case that is failing. Apart from that we can merge it.

**Pull request closed without merging**